src/Test/Desktop/OpenRiaServices.Common.DomainServices.Test/Mocks/MockDomainServices.cs

```diff
@@ -1598,12 +1598,41 @@ public void ThrowValidationException()
             throw new ValidationException("Validation error.");
         }
 
-        [Invoke]
-        public void VoidMethod()
+        [Invoke(HasSideEffects = true)]
+        [OutputCache(OutputCacheLocation.Client, 2)]
+        public void VoidInvokeWithSideEffectsAndCaching()
         {
 
         }
 
+        [Invoke(HasSideEffects = false)]
+        public void VoidInvokeNoSideEffectsAndNoCaching()
+        {
+
+        }
+
+        [Invoke(HasSideEffects = false)]
+        [OutputCache(OutputCacheLocation.Client, 2)]
+        public IEnumerable<CityWithCacheData> GetCitiesWithCachingInvoke()
+        {
+            return GetCities().AsEnumerable();
+        }
+
+        [Invoke(HasSideEffects = false)]
+        [OutputCache(OutputCacheLocation.Client, 2)]
+        public IEnumerable<CityWithCacheData> GetCitiesWithCachingAndThrowInvoke()
+        {
+            throw new InvalidOperationException(s_counter++.ToString());
+        }
+
+        // Cache on server because that's the only deterministic scenario.
+        [Invoke(HasSideEffects = false)]
+        [OutputCache(OutputCacheLocation.Server, 5, VaryByHeaders = "foo")]
+        public IEnumerable<CityWithCacheData> GetCitiesWithCachingVaryByHeadersInvoke()
```

Comment:
This method seems unused, I suggest it is removed if it is not used by any tests

Reply:
Removed